hydromt_wflow/wflow.py

```diff
@@ -761,8 +761,11 @@ def setup_gauges(
                 kwargs = {}
                 if isfile(gauges_fn):
                     # try to get epsg number directly, important when writting back data_catalog
-                    if self.crs.is_epsg_code:
-                        code = int(self.crs["init"].lstrip("epsg:"))
+                    if hasattr(self.crs, "to_epsg"):
+                        code = self.crs.to_epsg()
+                    elif hasattr(self.crs, "is_epsg_code"):
+                        if self.crs.is_epsg_code:
+                            code = int(self.crs["init"].lstrip("epsg:"))
```

Comment:
I think these lines can be removed because self.crs is always of pyproj.CRS type.

For the RasterModel see https://github.com/Deltares/hydromt/blob/075ac08d91f7e06bdc0504c0163d450e7b23a955/hydromt/models/model_grid.py#L181

For the Model see
https://github.com/Deltares/hydromt/blob/075ac08d91f7e06bdc0504c0163d450e7b23a955/hydromt/models/model_api.py#L1278

Reply:
This is true for hydromt >= 0.5.0 but not for older version I think. I don't mind removing I think we put pins everywhere already in the different env files.